subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -662,7 +662,7 @@ static int mod_sub_status(struct bt_mesh_model *model,
 		    (param->expect_sub && *param->expect_sub != sub_addr) ||
 		    param->cid != cid) {
 			BT_WARN("Model Subscription Status parameters did not match");
-			return -ENOENT;
+			goto done;
```

Comment:
```suggestion
			err = -ENOENT;
			goto done;
```

Reply:
and you need to declare `int err = 0;` in the local variable section.

Otherwise, we come back to what we started the discussion. This changes in function behavior and the function returns success even if the status has been failed.